plugins/modules/ndo_l3out_routed_interface.py

```diff
@@ -0,0 +1,544 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Akini Ross (@akinross) <akinross@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_l3out_routed_interface
+short_description: Manage L3Out Routed Interfaces on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Out Routed Interfaces on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Akini Ross (@akinross)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be an L3Out template.
+    - This parameter or O(template_id) is required.
+    type: str
+    aliases: [ l3out_template ]
+  template_id:
+    description:
+    - The ID of the L3Out template.
+    - This parameter or O(template) is required.
+    type: str
+    aliases: [ l3out_template_id ]
+  l3out:
+    description:
+    - The name of the L3Out.
+    - This parameter or O(l3out_uuid) is required.
+    type: str
+    aliases: [ l3out_name ]
+  l3out_uuid:
+    description:
+    - The UUID of the L3Out.
+    - This parameter or O(l3out) is required.
+    type: str
+  node_id:
+    description:
+    - The ID of the node (border leaf switch) where to deploy the L3Out routing protocol and node-level protocol configurations.
+    - This parameter is required when O(path) is specified.
+    - The node configuration is created under the L3Out template when it does not exist.
+    - The node configuration is updated under the L3Out template when it already exists.
+    - The node configuration is deleted under the L3Out template when there are no interfaces referencing it.
+    type: str
+    aliases: [ node, border_leaf ]
+  node_router_id:
+    description:
+    - The router ID of the node.
+    type: str
+    aliases: [ router_id ]
+  node_group_policy:
+    description:
+    - The name of the node group policy.
+    type: str
+  use_router_id_as_loopback:
+    description:
+    - Whether to use the router ID as the loopback address of the node.
+    type: bool
+  node_loopback_ip:
+    description:
+    - The loopback IP address of the node.
+    type: str
+    aliases: [ loopback_ip ]
+  path:
+    description:
+    - The path of the interface.
+    - The path is an existing physical port (eth1/1).
+    - This parameter or O(port_channel) is required.
```

Comment:
expand description for path/portchannel: https://github.com/CiscoDevNet/ansible-mso/pull/677/files#diff-3ac79155bcb61ba026579b561a90c07f4fe83063df886e899ce599d7be557130R119